app/views/products/index.html.haml

```diff
@@ -0,0 +1,50 @@
+.table-responsive
+  %table.table.table-striped
+    %thead
+      %tr
+        %th
+          = check_box_tag('check_all', 'check_all', false, onClick: "checkAll(this);")
+        %th
+          Name
+        %th
+          Image
+        %th
+          Vendor code
+        %th
+          Price
+        %th
+          Actions
+    %tbody
+      - @products.each do |product|
+        %tr
+          %td
+            = check_box_tag("check_#{product.id}", "check_#{product.id}", false, onClick: "checkOne(this);")
+          %td
+            = product.name
+          %td
+            -if product.image.attached? && product.image.persisted?
+              = image_tag product.image, size: "100x100", class: "img-fluid"
+            -else
+              = image_tag 'placeholder.jpg', size: "100x100", class: "img-fluid"
+          %td
+            = product.code
+          %td
+            = product.price
+          %td
+            = link_to 'View product', product_path(product), class: 'btn btn-outline-primary',
+            data: {turbolinks: false}
+            = link_to 'Update product', edit_product_path(product), class: 'btn btn-outline-warning',
```

Comment:
Let's use 'Edit product' for opening edit product page, and 'Update product' on real update action